### ISSUE TITLE
Which pretrained model to use for in-the-wild images?

### ISSUE THREAD

**Issue by janehwu** [SamsungLabs/HandNeRF#2]
I'm trying to extend `demo/demo.py` to work for any arbitrary input image (instead of the hard-coded `demo/input.jpg`). Should I still use the [demo pretrained HandNeRF model](https://drive.google.com/drive/folders/1AIYw8v8VlzlUk5dBEknYQQUq1DzVL29-?usp=sharing) or one of the [other released models](https://docs.google.com/spreadsheets/d/1j1GBITu8uwzR9To2ANgrRY5k8z_DeX0SNb-y_wCMP8Y/edit?usp=sharing)?

Also, are there any other variables in `demo.py` that need to be updated? For instance, the camera intrinsics are hard-coded, and `cfg.do_object_segmentation=True` does not appear in other scripts/pretrained models such as `main/recon.py`.

I tried using a DexYCB image from the training data, and the input data all looks correct:
<img src="https://github.com/SamsungLabs/HandNeRF/assets/9442165/ebf7306f-4fc1-4deb-8096-1fc8f5eef833" height="360">

But the output object looks like this:
<img src="https://github.com/SamsungLabs/HandNeRF/assets/9442165/a4ae2163-8ce3-44de-b28a-4f789ecc41de" height="360">
with the HandOccNet prediction:
<img src="https://github.com/SamsungLabs/HandNeRF/assets/9442165/e3a42922-1e0e-4c17-9a2a-2489aecaac55" height="360">

**Comment by hongsukchoi** [comment]
Hi!

If you want to test the model on any arbitrary image, it's recommended to use Table 3's HandNeRF model (25th row). The experiments in our paper use ground truth segmentation of an object as input, which can be replaced with off-the-shelf estimator's segmentation prediction. 

`cfg.do_object_segmentation=True` using the internal segmentation model. The demo model is only trained on one model, which combines the internal object segmentation model and HandNeRF.

Last, Table 3 HandNeRF's generalizability to unseen objects is limited. It is only trained on around 10 objects.

**Comment by janehwu** [comment]
Thanks so much for the pointers!

I was able to replace the demo model with Table 3's HandNeRF model, and I set `cfg.do_object_segmentation=False` since I can obtain segmentation masks from an off-the-shelf model.

However, I'm still trying to get the above^ DexYCB image working starting from `demo.py`. It looks like the remaining problem now is how I'm generating `world_3d_data`, specifically `world_3d_data['inside']`. I need to change what's currently in `demo.py` since the value is currently just set to all 1's (I assume because `cf.do_object_segmentation` was set to True): https://github.com/SamsungLabs/HandNeRF/blob/95cb4bfa4e0959955ec85bbaf5c4b44a0642e356/demo/demo.py#L146-L148

Could you explain these lines in `DexYCB.py`? https://github.com/SamsungLabs/HandNeRF/blob/c278b4d2286db4313749e6cfba63a51fca375884/data/DexYCB/DexYCB.py#L865-L871

It looks like the default is `cfg.use_multiview_masks_for_recon=True`, but I'm not sure if that means segmentation masks from multiple RGB images are used (which wouldn't be available at test time).

**Comment by hongsukchoi** [comment]
> I was able to replace the demo model with Table 3's HandNeRF model, and I set cfg.do_object_segmentation=False since I can obtain segmentation masks from an off-the-shelf model.

Yes, it's right!

> It looks like the default is cfg.use_multiview_masks_for_recon=True, but I'm not sure if that means segmentation masks from multiple RGB images are used (which wouldn't be available at test time).

The `cfg.use_multiview_masks_for_recon=True` is for efficiency. As you mentioned, multi-view RGB images are not available in inference time. You need to use only one backprojected segmentation mask as in the `demo.py`. Note that the segmentation mask is combination of hand and object mask.
